aospy/test/test_automate.py

```diff
@@ -183,48 +182,31 @@ def test_permute_core_specs(self, calc_suite):
         #       different default children for each
 
     def test_get_regions(self, calc_suite, all_regions):
-        assert set(calc_suite._get_regions()[0]) == set(all_regions)
+        assert calc_suite._get_regions()[0] == set(all_regions)
         # TODO: case w/ not all regions
         # TODO: case w/ Region objects in 'regions' sub-module
 
     def test_get_variables(self, calc_suite, all_vars):
         assert not hasattr(calc_suite, 'variables')
-        assert calc_suite._get_variables() == [condensation_rain,
-                                               convection_rain]
+        assert calc_suite._get_variables() == {condensation_rain,
+                                               convection_rain}
         # TODO: case w/ Var objects in 'variables' sub-module
         # TODO: case w/ 'all'
 
     def test_get_aux_specs(self, calc_suite, all_regions):
         spec_names = [name for name in calc_suite._AUX_SPEC_NAMES
                       if name not in [_VARIABLES_STR, _REGIONS_STR]]
         expected = {name: calc_suite._specs_in[name] for name in spec_names}
-        expected[_VARIABLES_STR] = calc_suite._specs_in[_VARIABLES_STR]
-        expected[_REGIONS_STR] = [all_regions]
+        expected[_VARIABLES_STR] = {condensation_rain, convection_rain}
+        expected[_REGIONS_STR] = [{globe, sahel}]
         expected['date_ranges'] = ['default']
         expected['output_time_regional_reductions'] = [['av', 'reg.av']]
         actual = calc_suite._get_aux_specs()
-        assert _compare_dicts_unordered(expected, actual)
+        assert actual == expected
 
     def test_permute_aux_specs(self, calc_suite, calcsuite_specs):
         expected = _permuted_dicts_of_specs(calcsuite_specs)
         actual = calc_suite._permute_aux_specs()
-        for act, exp in zip(actual, expected):
-            assert _compare_dicts_unordered(act, exp)
-
-
-def _compare_dicts_unordered(dict1, dict2):
-    """Key-by-key comparison of two dicts, ignoring orderingte of values."""
-    if dict1.keys() != dict2.keys():
-        return False
-    for key in dict1.keys():
-        # Check equality before using set() to account for strings.
-        if not dict1[key] == dict2[key]:
-            try:
-                set_equal = set(dict1[key]) == set(dict2[key])
-            # set() only accepts hashables
-            except TypeError:
-                return False
-            else:
-                if not set_equal:
-                    return False
-    return True
+        assert len(actual) == len(expected)
+        for act in actual:
+            assert act in expected
```

Comment:
But of course!  Much cleaner solution here.  Sorry for leading you down the wrong path.